server.py

```diff
@@ -91,6 +93,36 @@ def server_static(filepath):
     return static_file(filepath, root=os.path.join(dir_path, "static"))
 
 
+@app.get("/ranking/<code>")
+def contest_ranking(code):
+    with shelve.open(database_path) as submission_record:
+        order = [
+            (
+                user,
+                len(
+                    set(
+                        [
+                            attempt.question
+                            for attempt in submissions
+                            if (
+                                attempt.is_correct
+                                and (int(attempt.question) in contests[code].questions)
+                                and attempt.contest == code
+                                and attempt.time <= contests[code].end_time
+                                and attempt.time >= contests[code].start_time
+                            )
+                        ]
+                    )
+                ),
+            )
+            for user, submissions in submission_record.items()
+        ]
+    order.sort(key=lambda x: x[1], reverse=True)
+    order = [entry for entry in order if entry[1] > 0]
```

Comment:
@theSage21 can you suggest some way for me to remove line 121?
`order = [entry for entry in order if entry[1] > 0]`

Reply:
umm. you could combine both of them.
```python
order = [entry for entry in sorted(order, key=lambda x: x[1], reverse=True) if entry[1] > 0]
```

Reply:
another way is that when you are generating the order list don't append things that are  <= 0.
```python
order = [entry for entry in (
                                          (user, <some complex calculation)
                                          for something in something
                                         ) if entry[1] > 0]
order.sort(<fancy arguments>)
```
the inner iteration being `()` makes it a generator which means nothing is stored. Then in the outer loop, once the condition is met it is added to the list. So actually you don't add the entry if the condition is met. then you just sort it.

The current method is ok though. It's easy to read and works

Reply:
The contents of `order` look complicated already, so... I will go with the current one for now.

Reply:
Let me know if there is something else I can add to this PR